components/nexus-core/src/main/java/org/sonatype/nexus/internal/node/IdentityCapability.java

```diff
@@ -67,61 +56,12 @@ protected IdentityCapabilityConfiguration createConfig(final Map<String, String>
     return new IdentityCapabilityConfiguration(properties);
   }
 
-  public Certificate certificate() throws KeystoreException {
-    return keyStoreManager.getCertificate();
-  }
-
-  public String certificateAsPem() throws Exception {
-    return CertificateUtil.serializeCertificateInPEM(certificate());
-  }
-
-  public String certificateFingerprint() throws Exception {
-    return fingerprint;
-  }
-
-  @Override
-  protected void configure(final IdentityCapabilityConfiguration config) throws Exception {
-    // Generate identity key-pair if not already created
-    if (!keyStoreManager.isKeyPairInitialized()) {
-      log.info("Generating identity certificate");
-
-      // For now give something unique to the cert for additional identification purposes
-      UUID cn = UUID.randomUUID();
-      keyStoreManager.generateAndStoreKeyPair(
-          cn.toString(),
-          "Nexus",
-          "Sonatype",
-          "Silver Spring",
-          "MD",
-          "US");
-    }
-
-    Certificate cert = certificate();
-    log.trace("Certificate:\n{}", cert);
-
-    fingerprint = CertificateUtil.calculateFingerprint(cert);
-    log.debug("Fingerprint: {}", fingerprint);
-  }
-
-  @Override
-  protected void onActivate(final IdentityCapabilityConfiguration config) throws Exception {
-    // prime local node-id now
-    localNode.getId();
-  }
-
-  @Override
-  protected void onPassivate(final IdentityCapabilityConfiguration config) throws Exception {
-    localNode.reset();
-  }
-
-  @Override
-  protected void onRemove(final IdentityCapabilityConfiguration config) throws Exception {
-    if (keyStoreManager.isKeyPairInitialized()) {
-      log.debug("Clearing identity keys");
-      keyStoreManager.removePrivateKey();
-    }
-    fingerprint = null;
-  }
+  // FIXME: This does not actually work, will have to add some sort of hook/condition/magic
+  //@Override
+  //protected void onRemove(final IdentityCapabilityConfiguration config) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACommentedOutCodeLine)